### ISSUE TITLE
KeyboardAvoidingView flickering and hangs with Expo SDK 40, React Navigation, and react-native-screens

### ISSUE THREAD

**Issue by kevgrig** [expo/expo#11326]
## 🐛 Bug Report

Cross-posting this [really nasty bug](https://github.com/software-mansion/react-native-screens/issues/749) for awareness. It can cause freezes of not just the app but sometimes of the entire phone! https://github.com/software-mansion/react-native-screens/issues/749

Workaround seems to be:

```
//import { enableScreens } from 'react-native-screens';
//enableScreens();
```

![101942552-91af7a00-3b9e-11eb-8bc2-21f8739f6c14](https://user-images.githubusercontent.com/6432587/101944170-25824580-3ba1-11eb-957a-c315d856097a.gif)

**Comment by brentvatne** [comment]
thanks for sharing this! @kevgrig - one thing that might be helpful for the react-native-screens team to handle this more quickly and understand the root cause is to remove react-navigation and reproduce it only with react-native-screens

**Comment by kevgrig** [comment]
@brentvatne Interestingly, the issue only happens with react-navigation included. [The demo](https://github.com/kevgrig/rnkavflicker/blob/master/App.js) includes a flag (`const USE_REACT_NAVIGATION = true;`) because I tried to minimize the example further but it didn't happen with the same view but without react navigation (so I guess this might also be a bug in react navigation).

**Comment by brentvatne** [comment]
it could be related to react-navigation, but also the way that you're using it might not be the best. imo you should not wrap the navigation container with another view that modifies the layout of the navigationcontainer

**Comment by brentvatne** [comment]
closing this because it's some issue in other libraries

**Comment by kevgrig** [comment]
Interesting. Moving the `KeyboardAvoidingView` within `NavigationContainer` works. Is there a particular reason having `KeyboardAvoidingView` above the navigation container is an anti-pattern? I got tired to peppering them all over my app and preferred to just put it at the root (although now I know this, I can just make sure it's within `NavigationContainer` which is also in the root of my app).

**Comment by brentvatne** [comment]
there are practical concerns, like what if you had a screen where you wanted to have different behavior for the textinput? now you'd need to jump through all sorts of hoops or go back to adding it per-screen. then there are technical concerns, like re-rendering and updating the layout of your entire app because you add padding to the bottom of your root view which then forces a cascading relayout.

**Comment by kevgrig** [comment]
That makes sense, thanks!

**Comment by kevgrig** [comment]
Note for anyone else hitting this that, in testing my more complicated application, moving `KeyboardAvoidingView` inside of `NavigationContainer` did not help and commenting out the two `enableScreens` lines was still the only workaround.

**Comment by brentvatne** [comment]
@kevgrig - i would investigate any other components that you have in your application that wrap navigationcontainer

**Comment by kevgrig** [comment]
@brentvatne Yes, I do have some other wrapping components and they're likely driving this but I've spent quite a bit of time on this already so I'm just going to use the workaround of disabling screens for now. It seems like that's a performance optimization and my testing doesn't show a major degradation so I'll wait to see what the folks at software-mansion respond with.

**Comment by brentvatne** [comment]
i'd encourage you to try to make a more minimal example, reproducing the exact issue that you are seeing without involving react-navigation at all. the closer you can get to a minimal reproducible example, the easier it will be for the maintainers to respond and fix. the further it is, the more likely that the maintainers will close and assume it's related to another library.

**Comment by kevgrig** [comment]
@brentvatne I'll try to do that this weekend. I'm not convinced this can be reproduced without react navigation; in my more complicated app where the issue re-occurred even after moving `KeyboardAvoidingView` lower, that app still has react navigation. In my limited testing, the issue only occurs with react navigation (although commenting out `enableScreens` resolves the issue, so I think the initial location of the bug report in react-native-screens is still reasonable). It seems like this is driven by `enableScreens` plus something wrapping react navigation.

**Comment by brentvatne** [comment]
@kevgrig - we just released a patch version that includes the fix for this from react-native-screens. you can install react-native-screens@2.15.2 and install the latest version of the expo app to get the fix. you can know if you're on the latest version because it will be called ["Expo Go"](https://blog.expo.io/expo-go-a-new-name-for-the-expo-client-4684a2709904)

**Comment by kevgrig** [comment]
@brentvatne Sorry for the delay in trying this. We've been launching our app in both stores in the last two weeks (and it all went great! Thanks to Expo!). I finally got around to this and confirmed the issue is fixed. Thanks!
